internal/app/wwctl/container/build/main.go

```diff
@@ -12,6 +18,51 @@ func CobraRunE(cmd *cobra.Command, args []string) error {
 		Force:          BuildForce,
 		All:            BuildAll,
 		Default:        SetDefault,
+		Initramfs:      Initramfs,
+	}
+	if Initramfs {
+		return runInitramfsBuild(cmd, cbp)
 	}
 	return container.ContainerBuild(cbp)
 }
+
+func runInitramfsBuild(cmd *cobra.Command, cbp *wwapiv1.ContainerBuildParameter) (err error) {
+	// TODO here we need to bind dracut.module file
+	// cexec.SetBinds()
+	if cbp == nil {
+		return fmt.Errorf("ContainerBuildParameter is nill")
+	}
+
+	var containers []string
+	if cbp.All {
+		containers, err = pkgcontianer.ListSources()
+	} else {
+		containers = cbp.ContainerNames
+	}
+
+	if len(containers) == 0 {
+		return
+	}
+
+	for _, c := range containers {
+		// kernel version, we need to set container kernel version as by default, it'll build against
+		// host kernel version, which usually does not exist inside container
+		var kver string
+		rootfsDir := pkgcontianer.RootFsDir(c)
+		kver, err = kernel.FindKernelVersion(rootfsDir)
+		if err != nil {
+			return fmt.Errorf("failed to locate container kernel version: %s", err)
+		}
+
+		err = cexec.CobraRunE(cmd, []string{c, "/usr/bin/dracut --no-hostonly --force --verbose --kver " + kver})
+		if err != nil {
+			return
+		}
```

Comment:
I think that this is running dracut on the host, right? If we go down the dracut path, we have to run dracut within the container.

Reply:
Nope, it should run inside the container. This utilizes the 'container exec' sub-command. Check this file 'internal/app/wwctl/container/exec/child/main.go'